csharp/src/Apache.Arrow.Compression/Lz4CompressionCodec.cs

```diff
@@ -18,7 +18,7 @@
 using CommunityToolkit.HighPerformance;
 using K4os.Compression.LZ4.Streams;
 
-namespace Apache.Arrow.Tests.Compression
+namespace Apache.Arrow.Compression
 {
     internal sealed class Lz4CompressionCodec : ICompressionCodec
```

Comment:
Since the `Lz4CompressionCodec` doesn't need to be disposed, would it make sense to just have a singleton object and return that from `CreateCodec`? It would save on a tiny allocation every time this is called.

(I have the same comment for the `NoOpBufferCreator` in the previous PR, but I missed my window to add the comment there. Maybe we could update that as well?)

Reply:
Yes that would make sense, I will update both.